src/draive/types/model.py

```diff
@@ -1,35 +1,16 @@
 import json
+from dataclasses import asdict
 from typing import Self
 
-from draive.types.missing import MISSING
-from draive.types.parameters import ParameterSpecification
-from draive.types.state import State
+from draive.types.specification import ParametrizedModel
 
 __all__ = [
     "Model",
 ]
 
 
-class Model(State):
-    @classmethod
-    def specification(cls) -> ParameterSpecification:
-        if not hasattr(cls, "_specification"):
-            cls._specification: ParameterSpecification = {
-                "type": "object",
-                "properties": {
-                    parameter.alias or parameter.name: parameter.specification()
-                    for parameter in cls._parameters()
-                },
-                "required": [
-                    parameter.alias or parameter.name
-                    for parameter in cls._parameters()
-                    if parameter.default is MISSING
-                ],
-            }
-
-        return cls._specification
-
-    @classmethod
+class Model(ParametrizedModel):
+    @classmethod  # avoid making json each time
```

Comment:
For immutable objects you can add `@cached_property` decorator so they are only calculated once.

Reply:
it might not work properly with classmethods 🤔

Reply:
It was just a general comment.